server/src/com/thoughtworks/go/config/update/PatchEnvironmentCommand.java

```diff
@@ -74,15 +78,50 @@ public void update(CruiseConfig configForEdit) throws Exception {
         for (String pipelineName : pipelinesToRemove) {
             environmentConfig.removePipeline(new CaseInsensitiveString(pipelineName));
         }
+
+        for (EnvironmentVariableConfig variableConfig : envVarsToAdd) {
+            environmentConfig.addEnvironmentVariable(variableConfig);
+        }
+
+        for (String variableName : envVarsToRemove) {
+            environmentConfig.getVariables().removeIfExists(variableName);
+        }
     }
 
     @Override
     public boolean isValid(CruiseConfig preprocessedConfig) {
         boolean isValid = validateRemovePipelines(preprocessedConfig);
         isValid = isValid && validateRemoveAgents(preprocessedConfig);
```

Comment:
this can not be done, as the errors are populated into the result object. and validating everything will result into overriding the previous error message.